webapp/src/js/services/select2-search.js

```diff
@@ -137,6 +137,9 @@ angular.module('inboxServices').factory('Select2Search',
           } else {
             resolution = getDoc(value).then(function(doc) {
               selectEl.select2('data')[0].doc = doc;
+            })
+            .catch(err => {
+              $log.warn('Unable to hydrate select2 selection', err);
```

Comment:
I decided to suppress this error if the lineage fails because it is suppressed on line 191. But would like to confirm that this seems like reasonable fail-over behaviour for this service.

Reply:
Yes it's reasonable to treat both cases the same. I think it would be slightly better to modify the `getDoc` function to catch the error, check for a 404, and return `undefined`, eg:

```js
      const getDoc = id => {
        return LineageModelGenerator.contact(id, { merge: true })
          .then(contact => {
            const doc = contact.doc;
            doc.muted = ContactMuted(doc);
            return doc;
          })
          .catch(err => {
            if (err.code === 404) {
              $log.warn('something');
              return;
            }
            throw err;
          });
      };
```

Then the callers can set the selected value to null (I think that'll just work), and `$log.error` anything unexpected.

Reply:
Okay. We will use an undefined doc instead of just ignoring the error.